sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py

```diff
@@ -266,7 +320,308 @@ def get_long_running_output(pipeline_response):
+
+        def get_long_running_output(pipeline_response):
+            response_headers = {}
+            response = pipeline_response.http_response
+            response_headers["Retry-After"] = self._deserialize("int", response.headers.get("Retry-After"))
+            response_headers["Operation-Location"] = self._deserialize(
+                "str", response.headers.get("Operation-Location")
+            )
+
+            deserialized = _deserialize(_models.AnalyzeResult, response.json().get("analyzeResult"))
+            if cls:
+                return cls(pipeline_response, deserialized, response_headers)  # type: ignore
+            return deserialized
+
+        path_format_arguments = {
+            "endpoint": self._serialize.url("self._config.endpoint", self._config.endpoint, "str", skip_quote=True),
+        }
+
+        if polling is True:
+            polling_method: PollingMethod = cast(
+                PollingMethod, LROBasePolling(lro_delay, path_format_arguments=path_format_arguments, **kwargs)
+            )
+        elif polling is False:
+            polling_method = cast(PollingMethod, NoPolling())
+        else:
+            polling_method = polling
+        if cont_token:
+            return AnalyzeDocumentLROPoller[_models.AnalyzeResult].from_continuation_token(
+                polling_method=polling_method,
+                continuation_token=cont_token,
+                client=self._client,
+                deserialization_callback=get_long_running_output,
+            )
+        return AnalyzeDocumentLROPoller[_models.AnalyzeResult](
+            self._client, raw_result, get_long_running_output, polling_method  # type: ignore
+        )
+
+    @distributed_trace
+    def begin_analyze_batch_documents(
+        self,
+        model_id: str,
+        analyze_batch_request: Optional[Union[_models.AnalyzeBatchDocumentsRequest, JSON, IO[bytes]]] = None,
+        *,
+        pages: Optional[str] = None,
+        locale: Optional[str] = None,
+        string_index_type: Optional[Union[str, _models.StringIndexType]] = None,
+        features: Optional[List[Union[str, _models.DocumentAnalysisFeature]]] = None,
+        query_fields: Optional[List[str]] = None,
+        output_content_format: Optional[Union[str, _models.ContentFormat]] = None,
+        output: Optional[List[Union[str, _models.AnalyzeOutputOption]]] = None,
+        **kwargs: Any,
+    ) -> LROPoller[_models.AnalyzeBatchResult]:
+        return super().begin_analyze_batch_documents(
+            model_id=model_id,
+            analyze_batch_request=analyze_batch_request,
+            pages=pages,
+            locale=locale,
+            string_index_type=string_index_type,
+            features=features,
+            query_fields=query_fields,
+            output_content_format=output_content_format,
+            output=output,
+            polling=AnalyzeBatchDocumentsLROPollingMethod(timeout=self._config.polling_interval),
```

Comment:
We should accept a user-provided method override of `polling_interval` as well

Reply:
`self._config.polling_interval` can accept user-provided `polling_interval`, so it's not necessary to have an override here.

Reply:
agree it should be there, if the user has been sending it in. You can use the value from the config as default if its not there

Reply:
We should support overriding at client and method level. Because we are passing a custom polling method here, a user-passed `polling_interval` will be ignored in the generated code:

https://github.com/Azure/azure-sdk-for-python/blob/48755d89a2c1021ed6ad31d12182c78dd74a228a/sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py#L1339

We need to pop the kwarg ourselves

Reply:
Have added the method level passing `polling_interval`.